integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/BrushTest.kt

```diff
@@ -89,28 +91,33 @@ fun CustomBrushTest() {
                     repeatMode = RepeatMode.Reverse
                 )
         )
-
+    lateinit var brushState: State<Brush>
     // Android T introduces AGSL and RuntimeShader. Robolectric (part of our test infrastructure)
     // only supports software rendering.
-    val brush: () -> Brush =
-        if (
-            Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU &&
-                "robolectric" != Build.FINGERPRINT
-        ) {
-            val shader = RuntimeShader(CUSTOM_SHADER)
-            val shaderBrush = SizingShaderBrush(shader)
-            // The kotlin compiler seems to get confused without semicolons on these statements.
-            shader.setFloatUniform("iTime", 0.0f);
-
-            {
-                // Only sample the state in the generator function; this means that we avoid
-                // recomposition and only do the redraw phase.
-                shader.setFloatUniform("iTime", movingValue.value)
-                shaderBrush
+    if (
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU && "robolectric" != Build.FINGERPRINT
+    ) {
+        val shader = RuntimeShader(CUSTOM_SHADER)
+        val shaderBrush = SizingShaderBrush(shader)
+        // The kotlin compiler seems to get confused without semicolons on these statements.
+        shader.setFloatUniform("iTime", 0.0f)
+        brushState =
+            object : State<Brush> {
+                override val value: Brush
+                    get() =
+                        shaderBrush.also {
+                            // Only sample the state in the generator function; this means that we
```

Comment:
I don't recall what the "generator" function is, but I assume that basically when you access the value it calls this and updates the uniform.

Reply:
Yes.
It used to be: https://github.com/google/automotive-design-compose/blob/9ee3d5f24905ab093f6506106c6527e40da2837d/integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/examples/BrushTest.kt#L108